airflow/www/security.py

```diff
@@ -56,8 +57,11 @@
     "Public",
 }
 
+# Fetch the security manager override from the auth manager
+SecurityManagerOverride = get_auth_manager().get_security_manager_override_class()
 
-class AirflowSecurityManager(SecurityManager, LoggingMixin):
+
+class AirflowSecurityManager(SecurityManager, LoggingMixin, SecurityManagerOverride):  # type: ignore
```

Comment:
Instead of resorting to blanket ignore, I think we can do

```python
if TYPE_CHECKING:
    SecurityManagerOverride: type = object
else:
    SecurityManagerOverride = get_auth_manager().get_security_manager_override_class()
```

(or something like this, I think the order of conditionals or the `TYPE_CHECKING` conditional matters)

Reply:
Also does this order work? I seem to recall Python MRO is front over after, so in this setup `SecurityManager` would simply shadow anything declared in `SecurityManagerOverride`. This doesn’t seem to be what you intend.

```pycon
>>> class A:
...  def f(self): pass
... 
>>> class B:
...  def f(self): pass
... 
>>> class X(A, B): pass
... 
>>> class Y(B, A): pass
... 
>>> X.f
<function A.f at 0x104d9eac0>
>>> Y.f
<function B.f at 0x104d9eb60>
>>>
```

Reply:
So `SecurityManager` overriding `SecurityManagerOverride` is intended? In that case the base class should not be named `SecurityManagerOverride` IMO.

Reply:
Thanks! I did not know that. `SecurityManagerOverride` should indeed take precedence over `SecurityManager`. I fixed the order